### PR TITLE
fix: validate rules in loop

### LINKED ISSUE #53: Setup CI/CD to the repository
For examples see: https://github.com/redhat-developer/app-services-sdk-go
makefile should define steps

### LINKED ISSUE #93: CI/CD support for validator
Run tests and compile binary with PR

### LINKED ISSUE #98: Rule methods should implement interface
To make it easy for adding more custom rules to validator, rules methods should implement an interface. 
we should have the same interface and the same function. Without that there will be no way to add rule to validator. This two methods should be the same method

### Currently
lengthRule implements - `validateLength`
MustExistRule implements - `validateMustExist`

### PR DESCRIPTION
**Pull request by ankithans**
Closes #98 
Closes #93 
Closes #53 

### Description
- Rules implement `Rule` interface
- Validate rules in for loop(DRY), instead of calling one by one
- CI/CD tests & builds validator example

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: validator/rules/executor.go [hunk 1]
```diff
@@ -10,6 +10,10 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type Rules interface {
```
Comment (wtrocki) [inline validator/rules/executor.go:13]: doc

File: validator/rules/executor.go [hunk 2]
```diff
@@ -10,6 +10,10 @@ import (
...
+	Validate() []validator.ValidationError
```
Comment (wtrocki) [inline validator/rules/executor.go:14]: Wrap error aray in result structure for other elements

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
Lets release this /create github release v0.1.0

File: validator/rules/executor.go (hunk 2 above)
Comment (wtrocki) [reply validator/rules/executor.go:14]: actually lets leave it likt thid

**Comment by ankithans** [comment]
Draft release. Changes required?
https://github.com/aerogear/charmil/releases/tag/untagged-5c6b0883d263545a9332

**Pull request merged**